### PR TITLE
Ubuntu 22 04

### PR DESCRIPTION
**Pull request by maximiliankolb**
This PR aims to provide some documentation to provision hosts running Ubuntu 22.04.
See individual commits.

Cherry-pick into:

* [x] Foreman 3.3
* [x] Foreman 3.2
* [x] Foreman 3.1

### CODE REVIEW

File: guides/common/modules/proc_provisioning-ubuntu-20-04-3-plus-through-smart-proxies.adoc [hunk 1]
```diff
@@ -0,0 +1,36 @@
+[id="Provisioning_Ubuntu_20_04_3_plus_Through_Smart_Proxies_{context}"]
+= Provisioning Ubuntu 20.04.3+ Through {SmartProxies}
```
Comment (maximiliankolb) [inline guides/common/modules/proc_provisioning-ubuntu-20-04-3-plus-through-smart-proxies.adoc:2]: Do you like this title @bastian-src ? Should it rather include "Ubuntu Autoinstall"?

File: guides/common/modules/proc_provisioning-ubuntu-20-04-3-plus-through-smart-proxies.adoc [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+.Procedure
+. On your {SmartProxyServer}, ensure _Smart Proxy HTTP forward_ is enabled:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# grep "^:enabled: true$" /etc/foreman-proxy/settings.d/templates.yml
+----
++
+Note that this is enabled by default.
+. On your {SmartProxyServer}, open the corresponding port for the Smart Proxy service:
```
Comment (maximiliankolb) [inline guides/common/modules/proc_provisioning-ubuntu-20-04-3-plus-through-smart-proxies.adoc:15]: I am unsure if we can/should say "Smart Proxy service" here.

**Review by maximiliankolb — COMMENTED** [review]
Ready for review @bastian-src

File: guides/common/modules/proc_provisioning-ubuntu-20-04-3-plus-through-smart-proxies.adoc (hunk 2 above)
Comment (bastian-src) [reply guides/common/modules/proc_provisioning-ubuntu-20-04-3-plus-through-smart-proxies.adoc:15]: I get what you mean, service is a term which makes me think more about a daemon and that wouldn't fit too well here.
Instead, we could say:
```suggestion
. On your {SmartProxyServer}, open the corresponding port so the Smart Proxy can process incoming requests.
```

File: guides/common/modules/proc_provisioning-ubuntu-20-04-3-plus-through-smart-proxies.adoc (hunk 1 above)
Comment (bastian-src) [reply guides/common/modules/proc_provisioning-ubuntu-20-04-3-plus-through-smart-proxies.adoc:2]: Yes, we should say:
```suggestion
= Provisioning Ubuntu Autoinstall Through {SmartProxies}
```
Since you mention the specific versions in the short notice beneath, we don't need the version here.

File: guides/common/modules/proc_creating-an-installation-medium-for-ubuntu-22-04.adoc [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+# cd /tmp/
+# wget https://releases.ubuntu.com/22.04/ubuntu-22.04.1-live-server-amd64.iso
+----
+. Mount the ISO image:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# mount ubuntu-22.04.1-live-server-amd64.iso /mnt
+----
+. Provide both the ISO image and the extracted directory under _{foreman-example-com}/pub_:
```
Comment (bastian-src) [inline guides/common/modules/proc_creating-an-installation-medium-for-ubuntu-22-04.adoc:26]:
```suggestion
. Provide both, the ISO image and the extracted directory under _{foreman-example-com}/pub_:
```

Comment (maximiliankolb) [reply guides/common/modules/proc_creating-an-installation-medium-for-ubuntu-22-04.adoc]: Removed "both" instead.

File: guides/common/modules/proc_provisioning-ubuntu-20-04-3-plus-through-smart-proxies.adoc (hunk 1 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_provisioning-ubuntu-20-04-3-plus-through-smart-proxies.adoc]: Applied.

File: guides/common/modules/proc_provisioning-ubuntu-20-04-3-plus-through-smart-proxies.adoc (hunk 2 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_provisioning-ubuntu-20-04-3-plus-through-smart-proxies.adoc]: Removed "for the Smart Proxy service" instead.

**Comment by maximiliankolb** [comment]
I've applied all feedback by Bastian. Do you mind having a look at it @Lennonka ?

File: guides/common/modules/proc_provisioning-ubuntu-autoinstall-through-smart-proxies.adoc [hunk 4]
```diff
@@ -0,0 +1,36 @@
+[id="Provisioning_Ubuntu_Autoinstall_Through_Smart_Proxies_{context}"]
+= Provisioning Ubuntu Autoinstall Through {SmartProxies}
+
+Perform these steps to provision hosts running Ubuntu 20.04.3+ or Ubuntu 22.04.
```
Comment (Lennonka) [inline guides/common/modules/proc_provisioning-ubuntu-autoinstall-through-smart-proxies.adoc:4]: Maybe `Ubuntu 20.04.3 and newer, or Ubuntu 22.04`

File: guides/common/modules/proc_provisioning-ubuntu-autoinstall-through-smart-proxies.adoc [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+# firewall-cmd --reload
+----
+. On your {ProjectServer}, add your {SmartProxyServer} to the list of trusted proxies:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# {foreman-installer} \
+--foreman-trusted-proxies "127.0.0.1" \
+--foreman-trusted-proxies "::1" \
+--foreman-trusted-proxies "_Your_{SmartProxyServer}_IP_"
```
Comment (Lennonka) [inline guides/common/modules/proc_provisioning-ubuntu-autoinstall-through-smart-proxies.adoc:29]: The underscores are not working well together with the value of the variable, which contains spaces. Maybe use spaces instead, and maybe rephrase, like: `"_IP of your {SmartProxyServer}_"` ? (The most important info first, but it doesn't fit the convetion for replaceable values we use elsewhere...)
Or just leave out the variable and say something like `"_proxy_IP_"`?

File: guides/common/modules/proc_associating-templates-with-operating-systems.adoc [hunk 6]
```diff
@@ -0,0 +1,33 @@
+[id="Associating_Templates_with_Operating_Systems_{context}"]
+= Associating Templates with Operating Systems
+
+You can associate templates with operating systems in {Project}.
```
Comment (Lennonka) [inline guides/common/modules/proc_associating-templates-with-operating-systems.adoc:4]: Maybe mention in the intro why this is beneficial?

**Review by Lennonka — APPROVED** [review]
Just a couple of tiny suggestions for your consideration, otherwise LGTM!

File: guides/common/modules/proc_provisioning-ubuntu-autoinstall-through-smart-proxies.adoc (hunk 5 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_provisioning-ubuntu-autoinstall-through-smart-proxies.adoc]: Applied!

**Comment by maximiliankolb** [comment]
Rebased to master; merging after the GHA is green.

**Comment by maximiliankolb** [comment]
Merged to master and cherry-picked to 3.4: ✅ 
   98db4e564..746a6c475  3.4 -> 3.4
